### PR TITLE
Run the jaeger-agent as a non-root user by default

### LINKED ISSUE #2460: Run the jaeger-agent as a non-root user by default
## Requirement - what kind of business use case are you trying to solve?

I am trying to run the jaeger-agent container in a Kubernetes cluster with strict Pod Security Policies set, which requires all containers to run as non-root users.

## Problem - what in Jaeger blocks you from solving the requirement?

The Jaeger-agent container runs as root by default (https://github.com/jaegertracing/jaeger/blob/master/cmd/agent/Dockerfile).

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Update the Dockerfile to explicitely set a non-root `USER`. There's also some open discussion about this over at the Helm chart repo: https://github.com/jaegertracing/helm-charts/issues/41

I would suggest adding a USER directive to the end of the Dockerfile, setting the running user id to `10001`. I can create a PR for this. 

## Any open questions to address

* Should a different UID be used by default? - I prefer 10001 to 1001 suggested in https://github.com/jaegertracing/helm-charts/issues/41 as the latter UID is often already used by some account on the host machine.

### PR DESCRIPTION
**Pull request by chgl**
## Which problem is this PR solving?
- Resolves #2460 

## Short description of the changes
- Added a UID arg, defaulting to 10001, which the build image is running as by default.

### CODE REVIEW

File: cmd/agent/Dockerfile [hunk 1]
```diff
@@ -3,8 +3,11 @@ RUN apk add --update --no-cache ca-certificates
...
 FROM scratch
 ARG TARGETARCH=amd64
+ARG USER_UID=10001
 COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 
 EXPOSE 5775/udp 6831/udp 6832/udp 5778
 COPY agent-linux-$TARGETARCH /go/bin/agent-linux
 ENTRYPOINT ["/go/bin/agent-linux"]
+
+USER ${USER_UID}
```
Comment (yurishkuro) [inline cmd/agent/Dockerfile:13]: Just curious, does this UID have a name? How will it show in `ps`?

**Review by yurishkuro — APPROVED** [review]

File: cmd/agent/Dockerfile (hunk 1 above)
Comment (chgl) [reply cmd/agent/Dockerfile:13]: If there's no name/passwd entry associated with the UID, then the process' user should appear as `unknown` in `ps`. If you run kubectl exec, bash will usually show a prompt like `I have no name!@hostname:`. 

The advantage of using a numeric user vs an actual name is that if a Pod Security Policy in Kubernetes requires a container to be run as root (`runAsNonRoot: true`) then setting any "user" other than 0 here allows the container to be run without having to override the user id in the securityContext.

Comment (jpkrohling) [reply cmd/agent/Dockerfile:13]: For completeness: Kubernetes platforms like OpenShift would typically assign a non-root user dynamically anyway. I expect the same to be true for GKE and AKS.

**Comment by jpkrohling** [comment]
crossdock and unit tests CI jobs restarted.

**Comment by jpkrohling** [comment]
CI looks good. @chgl, could you please rebase this? Once it's done, I'll merge this one.

**Comment by jpkrohling** [comment]
crossdock-otel job restarted

**Comment by jpkrohling** [comment]
Thanks!

**Pull request merged**